### PR TITLE
[BugFix] show_data rest api support cloudnative table

### PR DESCRIPTION
**Pull request by kevincai**
Fixes #issue

## What type of PR is this:
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.1
  - [X] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowDataAction.java|0|1|00.00%|[70]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowDataAction.java|0|1|00.00%|[70]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowDataAction.java|0|1|00.00%|[70]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowDataAction.java|1|1|100.00%|[]|

**Review by starrocks-xupeng — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/http/rest/ShowDataAction.java [hunk 1]
```diff
@@ -68,7 +67,7 @@ public long getDataSizeOfDatabase(Database db) {
         // sort by table name
         List<Table> tables = db.getTables();
         for (Table table : tables) {
-            if (table.getType() != TableType.OLAP) {
+            if (!table.isNativeTableOrMaterializedView()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/ShowDataAction.java:70]: unlock in finally

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/ShowDataAction.java]: ok

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/ShowDataAction.java]: done

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowDataAction.java|7|8|87.50%|[72]|

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**